inception-workload/src/main/java/de/tudarmstadt/ukp/inception/workload/settings/WorkloadSettingsPanel.java

```diff
@@ -28,67 +30,97 @@
 
-        //Basic form
+        // Basic form
         Form<Void> form = new Form<>("form");
 
+        // Dropdown menu
         workloadStrategy = new BootstrapSelect<>("workloadStrategy");
-        workloadStrategy.setChoiceRenderer(
-                new LambdaChoiceRenderer<>(WorkloadManagerType::getUiName));
+        workloadStrategy
+                .setChoiceRenderer(new LambdaChoiceRenderer<>(WorkloadManagerType::getUiName));
+        workloadStrategy.setModel(LoadableDetachableModel.of(this::getWorkloadManager));
         workloadStrategy.setRequired(true);
         workloadStrategy.setNullValid(false);
         workloadStrategy.setChoices(workloadManagerExtensionPoint.getTypes());
-        workloadStrategy.setModel(LoadableDetachableModel.of(this::getWorkloadManager));
 
-        //add them to the form
+        // add them to the form
         form.add(workloadStrategy);
 
-        //Finally, add the confirm button at the end
+        // Finally, add the confirm button at the end
         Button confirm = new LambdaAjaxButton<>("save", this::actionConfirm);
         form.add(confirm);
 
         add(form);
     }
-    
+
+    /**
+     * This method returns the current WorkloadManagerType.
+     */
     private WorkloadManagerType getWorkloadManager()
     {
         WorkloadManager manager = workloadManagementService
-                .getOrCreateWorkloadManagerConfiguration(project);
-        WorkloadManagerExtension extension = workloadManagerExtensionPoint
-                .getExtension(manager.getType());
-        return new WorkloadManagerType(extension.getId(), extension.getLabel());
+                .loadOrCreateWorkloadManagerConfiguration(project);
+        Extension extension = workloadManagerExtensionPoint.getExtension(manager.getType());
+        return new WorkloadManagerType(extension.getId(), extension.getId());
     }
 
-    private void actionConfirm(AjaxRequestTarget aTarget, Form<?> aForm)
+    /**
+     * Confimation action of the button
+     */
+    private void actionConfirm(AjaxRequestTarget aTarget, Form<?> aForm) throws IOException
     {
         aTarget.addChildren(getPage(), IFeedback.class);
 
-        workloadManagementService.setWorkloadManagerConfiguration(
-                workloadStrategy.getModelObject().getWorkloadManagerExtensionId(), project);
-
-        success("Workload settings changed");
+        // Either traits are already available for the project
+        WorkloadManager manager = workloadManagementService
+                .loadOrCreateWorkloadManagerConfiguration(project);
+        if (manager != null) {
+            manager.setType(workloadStrategy.getModelObject().getWorkloadManagerExtensionId());
+            workloadManagementService.saveConfiguration(manager);
+        }
+        // or they are not available (first time after project creation or "older" projects)
+        else {
+            DynamicWorkloadTrait trait = new DynamicWorkloadTrait();
```

Comment:
New traits objects should be obtained from the `WorkloadExtension` for the selected type of workload. You should not have to call their constructors directly. Typically, this happens implicitly when you call `workloadManagerExtension.readTraits(workloadManager)` on workload manager that does not have any traits yet - the `readTraits()` would then create the traits object appropriate for the respective `workloadManagerExtension` implementation (e.g. the DynamicWorkloadExtension would create a DynamicWorkloadTraits`(typo "Trait" -> "Traits").